### ISSUE TITLE
pls fix 

### ISSUE THREAD

**Issue by ZZIDZZ** [Reylyer/truk-isekai#1]
This [Github Repo](https://www.youtube.com/watch?autoplay=1&v=c9bUJ6drLbk&feature=youtu.be) tells me that glut is already deprecated, but what about this?

GLFW is an Open Source, multi-platform library for OpenGL, OpenGL ES and Vulkan development on the desktop. It provides a simple API for creating windows, contexts and surfaces, receiving input and events.
GLFW is written in C and supports Windows, macOS, X11 and Wayland.
GLFW is licensed under the [zlib/libpng license](https://www.glfw.org/license.html).
Download GLFW 3.3.7
Released on April 5, 2022

GLFW 3.4 progress
50%
Gives you a window and OpenGL context with just [two function calls](https://www.glfw.org/documentation.html#example-code)
Support for OpenGL, OpenGL ES, Vulkan and related options, flags and extensions
Support for multiple windows, multiple monitors, high-DPI and gamma ramps
Support for keyboard, mouse, gamepad, time and window event input, via polling or callbacks

Comes with a [tutorial](https://www.glfw.org/docs/latest/quick.html), guides and reference [documentation](https://www.glfw.org/docs/latest/), examples and test programs

Open Source with an OSI-certified license allowing commercial use

Access to native objects and compile-time options for platform specific features

Community-maintained [bindings](https://www.glfw.org/community.html#bindings) for many different languages
No library can be perfect for everyone. If GLFW isn’t what you’re looking for, there are [alternatives]

**Comment by Reylyer** [comment]
You are right but also wrong, this repo uses freeglut, a superior version to glut that is still maintained.

freeglut is a free-software/open-source alternative to the OpenGL Utility Toolkit (GLUT) library. GLUT was originally written by Mark Kilgard to support the sample programs in the second edition OpenGL 'RedBook'. Since then, GLUT has been used in a wide variety of practical applications because it is simple, widely available and highly portable.

GLUT (and hence freeglut) takes care of all the system-specific chores required for creating windows, initializing OpenGL contexts, and handling input events, to allow for trully portable OpenGL programs.

freeglut is released under the X-Consortium license.

The original GLUT library seems to have been abandoned with the most recent version (3.7) dating back to August 1998. Its license does not allow anyone to distribute modified library code. This is really unfortunate, since GLUT is getting old and really needs improvement. Also, GLUT's license is incompatible with some software distributions (e.g., XFree86).

freeglut was originally written by Pawel W. Olszta with contributions from Andreas Umbach and Steve Baker.

John F. Fay, John Tsiombikas, and Diederick C. Niehorster are the current maintainers of the freeglut project.

Pawel started freeglut development on December 1st, 1999. The project is now virtually a 100% replacement for the original GLUT with only a few departures (such as the abandonment of SGI-specific features such as the Dials&Buttons box and Dynamic Video Resolution) and a shrinking set of bugs.

freeglut adds some additional features over the basic GLUT functionality, such as a larger set of predefined objects to use, the ability to run single iterations of the event loop, or exit from it gracefully, mousewheel input callbacks, optional OpenGL core/compatibility profile context creation, multitouch/multi-pointer input, and support for a larger and growing set of platforms, being just some of them.

**Comment by ZZIDZZ** [comment]
Yes, but have you ever realized that
FreeGLUT is an [open-source](https://en.wikipedia.org/wiki/Open-source_software) alternative to the [OpenGL Utility Toolkit](https://en.wikipedia.org/wiki/OpenGL_Utility_Toolkit) (GLUT) library. GLUT (and hence FreeGLUT) allows the user to create and manage windows containing OpenGL contexts on a wide range of platforms and also read the mouse, keyboard and joystick functions. FreeGLUT is intended to be a full replacement for GLUT, and has only a few differences.

Since GLUT has gone into stagnation, FreeGLUT is in development to improve the toolkit. It is released under the [MIT License](https://en.wikipedia.org/wiki/MIT_License).


Contents
1	History
1.1	OpenGLUT
2	Status
3	See also
4	References
5	External links
History
FreeGLUT was originally written by Paweł W. Olszta with contributions from Andreas Umbach and Steve Baker. Since Paweł ceased working in 3D graphics, he passed the baton to Steve Baker. The current maintainers of FreeGLUT are John F. Fay, John Tsiombikas, and Diederick C. Niehorster.

Paweł started FreeGLUT development on December 1, 1999. The project is now virtually a 100% replacement for the original GLUT with only a few departures (such as the abandonment of SGI-specific features such as the Dials&Buttons box and Dynamic Video Resolution) and a shrinking set of bugs.

FreeGLUT contains a few enhancements over the original GLUT - but as a matter of policy, no further significant features will be added.

In April 2015, FreeGLUT obtained initial support for the new Wayland display server protocol.[1]
